parquet-variant/src/main/java/org/apache/parquet/variant/VariantBuilder.java

```diff
@@ -101,6 +132,32 @@ public Variant build() {
     return new Variant(Arrays.copyOfRange(writeBuffer, 0, writePos), metadata);
   }
 
+  // This is used in the shredded reader to check if anything has been written to the Variant since the last call.
+  // Tracking in the reader code is a bit awkward, but maybe better than polluting this interface.
+  int getWritePos() {
+    return writePos;
+  }
+
+  /**
+   * @return the constructed Variant value binary, without metadata.
+   */
+  public byte[] valueWithoutMetadata() {
```

Comment:
I wonder if we should make metadata part of this API or if we should pass in a metadata instance that does the lookup to resolve a name. Then metadata would be a separate builder that implements that API.